sourcemap/fleet_store_test.go

```diff
@@ -77,4 +77,62 @@ func TestFleetFetch(t *testing.T) {
 	assert.True(t, hasAuth)
 }
 
+func TestMultipleFleetHostsFetch(t *testing.T) {
+	var (
+		hasAuth       bool
+		apikey        = "supersecret"
+		name          = "webapp"
+		version       = "1.0.0"
+		path          = "/my/path/to/bundle.js.map"
+		c             = http.DefaultClient
+		sourceMapPath = "/api/fleet/artifact"
+	)
+
+	ts0 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, sourceMapPath, r.URL.Path)
+		auth := r.Header.Get("Authorization")
+		hasAuth = auth == "ApiKey "+apikey
+		// zlib compress
+		wr := zlib.NewWriter(w)
+		defer wr.Close()
+		wr.Write([]byte(resp))
+	}))
+	defer ts0.Close()
+
+	ts1 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, sourceMapPath, r.URL.Path)
+		auth := r.Header.Get("Authorization")
+		hasAuth = auth == "ApiKey "+apikey
+		// zlib compress
+		wr := zlib.NewWriter(w)
+		defer wr.Close()
+		wr.Write([]byte(resp))
+	}))
+	defer ts1.Close()
```

Comment:
seeing as the handler is exactly the same in each case, how about we extract it? i.e. something like:

```go
h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
	assert.Equal(t, sourceMapPath, r.URL.Path)
	auth := r.Header.Get("Authorization")
	hasAuth = auth == "ApiKey "+apikey
	// zlib compress
	wr := zlib.NewWriter(w)
	defer wr.Close()
	wr.Write([]byte(resp))
})

ts0 := httptest.NewServer(h)
defer ts0.Close()

ts1 := httptest.NewServer(h)
defer ts1.Close()
```

Reply:
Having said that, we should probably make multiple requests and check which server served the request, to show that the hosts are randomly selected.

Reply:
I'm not clear on why it would be necessary to test that a host is selected randomly. I'm using a package function `rand.Perm`, which I can assume works correctly. So if we get a successful response, we can assume we are selecting a host randomly. Let me know if there's something important I'm missing about the way I've written the code.